aten/src/ATen/native/Unique.cpp

```diff
@@ -101,19 +146,21 @@ std::tuple<Tensor, Tensor, Tensor> _unique_dim_cpu_template(
   scalar_t* input_flat_ptr = ((scalar_t*)input_flat.data_ptr());
 
   // sort indices using data
-  std::sort(indices.begin(), indices.end(),
-    [&](int64_t a, int64_t b) -> bool {
-      for (int64_t i = 0; i < numel; ++i) {
-        scalar_t lhs = input_flat_ptr[i + a * numel];
-        scalar_t rhs = input_flat_ptr[i + b * numel];
-        if (lhs < rhs) {
-          return true;
-        } else if (lhs > rhs) {
-          return false;
+  if (!sorted_input) {
+    std::sort(indices.begin(), indices.end(),
+      [&](int64_t a, int64_t b) -> bool {
+        for (int64_t i = 0; i < numel; ++i) {
+          scalar_t lhs = input_flat_ptr[i + a * numel];
+          scalar_t rhs = input_flat_ptr[i + b * numel];
+          if (lhs < rhs) {
+            return true;
+          } else if (lhs > rhs) {
+            return false;
+          }
         }
-      }
-      return false;
-    });
+        return false;
+      });
+  }
 
   Tensor input_sorted = at::empty(input_flat.sizes(), input_flat.options());
   for (int i = 0; i < indices.size(); ++i) {
```

Comment:
Maybe just clone if it is already sorted?

Reply:
Actually no need to clone at all, just assign would work